code/modules/unit_tests/missing_icons.dm

```diff
@@ -159,6 +159,8 @@
 					for(var/obj/item/weapon/gun/guntype as anything in guntypes)
 						if(isnull(initial(guntype.icon_state)))
 							continue
+						if(!guntype.map_specific_decoration)
+							prefix = ""
 						check(obj_path, 'icons/obj/items/clothing/belts/holstered_guns.dmi', prefix + initial(guntype.icon_state), guntype, "gun_underlay")
```

Comment:
Sorry to change my mind on how to correct this, but currently resetting the prefix would mean that after the first gun doesn't have a map_specific_decoration in the loop, all subsequent checks will be without a prefix.

```suggestion
						if(!guntype.map_specific_decoration)
							check(obj_path, 'icons/obj/items/clothing/belts/holstered_guns.dmi', initial(guntype.icon_state), guntype, "gun_underlay")
						else
							check(obj_path, 'icons/obj/items/clothing/belts/holstered_guns.dmi', prefix + initial(guntype.icon_state), guntype, "gun_underlay")
```